plugins/aws/resoto_plugin_aws/resource/elasticache.py

```diff
@@ -0,0 +1,471 @@
+    cluster_engine: Optional[str] = field(default=None)
+    cluster_engine_version: Optional[str] = field(default=None)
+    cluster_cache_cluster_status: Optional[str] = field(default=None)
+    cluster_num_cache_nodes: Optional[int] = field(default=None)
+    cluster_preferred_availability_zone: Optional[str] = field(default=None)
+    cluster_preferred_outpost_arn: Optional[str] = field(default=None)
+    cluster_preferred_maintenance_window: Optional[str] = field(default=None)
+    cluster_pending_modified_values: Optional[AwsElastiCachePendingModifiedValues] = field(default=None)
+    cluster_notification_configuration: Optional[AwsElastiCacheNotificationConfiguration] = field(default=None)
+    cluster_cache_security_groups: List[AwsElastiCacheCacheSecurityGroupMembership] = field(factory=list)
+    cluster_cache_parameter_group: Optional[AwsElastiCacheCacheParameterGroupStatus] = field(default=None)
+    cluster_cache_subnet_group_name: Optional[str] = field(default=None)
+    cluster_cache_nodes: List[AwsElastiCacheCacheNode] = field(factory=list)
+    cluster_auto_minor_version_upgrade: Optional[bool] = field(default=None)
+    cluster_security_groups: List[AwsElastiCacheSecurityGroupMembership] = field(factory=list)
+    cluster_replication_group_id: Optional[str] = field(default=None)
+    cluster_snapshot_retention_limit: Optional[int] = field(default=None)
+    cluster_snapshot_window: Optional[str] = field(default=None)
+    cluster_auth_token_enabled: Optional[bool] = field(default=None)
+    cluster_auth_token_last_modified_date: Optional[datetime] = field(default=None)
+    cluster_transit_encryption_enabled: Optional[bool] = field(default=None)
+    cluster_at_rest_encryption_enabled: Optional[bool] = field(default=None)
+    cluster_replication_group_log_delivery_enabled: Optional[bool] = field(default=None)
+    cluster_log_delivery_configurations: List[AwsElastiCacheLogDeliveryConfiguration] = field(factory=list)
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(
+            service=self.api_spec.service, action="delete_cache_cluster", result_name=None, CacheClusterId=self.id
+        )
+        return True
+
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def add_tags(resource: AwsElastiCacheCacheCluster) -> None:
+            tags = builder.client.list(
+                resource.api_spec.service,
+                "list-tags-for-resource",
+                None,
+                ResourceName=resource.arn,
+            )
+            if tags:
+                resource.tags = bend(S("TagList", default=[]) >> ToDict(), tags[0])
+
+        for js in json:
+            instance = cls.from_api(js)
+            builder.add_node(instance, js)
+            builder.submit_work(add_tags, instance)
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        for sg in self.cluster_security_groups:
```

Comment:
I think you also want `super().connect_in_graph(builder, source)` here and then additionally the edge to the Security Group. If you do it like this, you *only* get the latter. Applies to the the replication group further down also